openedx/core/djangoapps/content/course_overviews/models.py

```diff
@@ -387,14 +387,8 @@ def get_from_id(cls, course_id):
         # Regenerate the thumbnail images if they're missing (either because
         # they were never generated, or because they were flushed out after
         # a change to CourseOverviewImageConfig.
-        if course_overview:
-            if hasattr(course_overview, 'image_set'):
-                image_set = course_overview.image_set
-                if not image_set.small_url or not image_set.large_url:
-                    CourseOverviewImageSet.objects.filter(course_overview=course_overview).delete()
-                    CourseOverviewImageSet.create(course_overview)
-            else:
-                CourseOverviewImageSet.create(course_overview)
+        if course_overview and not hasattr(course_overview, 'image_set'):
+            CourseOverviewImageSet.create(course_overview)
```

Comment:
I'm actually not even sure we need this? The ImageSet for a course is recreated every time the course is published in studio. So it wouldn't normally hit this path, unless (1) maybe you just enabled the image set creation config and want to lazily create these at request-time rather than publish-time? do we care about that distinction? or (2) something went wrong during publish and we want to ensure that `image_set` is defined. Which.... OK

Reply:
I don't know enough to have an opinion either way. This approach could also have maybe just been a backwards compatibility support method?
But assuming you keep it, this looks fine to me.